std/string.d

```diff
@@ -2033,6 +2040,211 @@ S[] splitLines(S)(S s, in KeepTerminator keepTerm = KeepTerminator.no) @safe pur
     });
 }
 
+/***********************************
+ *  Split an array or slicable range of characters into a range of lines
+    using $(D '\r'), $(D '\n'),
+    $(D "\r\n"), $(XREF uni, lineSep), and $(XREF uni, paraSep) as delimiters.
+    If $(D keepTerm) is set to $(D KeepTerminator.yes), then the delimiter
+    is included in the slices returned.
+
+    Does not throw on invalid UTF; such is simply passed unchanged
+    to the output.
+
+    Does not allocate memory.
+
+  Params:
+    r = array of $(D chars), $(D wchars), or $(D dchars) or a slicable range
+    keepTerm = whether delimiter is included or not in the results
+  Returns:
+    range of slices of the input range $(D r)
+
+  See_Also:
+    $(LREF splitLines)
+    $(XREF algorithm, splitter)
+    $(XREF regex, splitter)
+ */
+auto splitterLines(KeepTerminator keepTerm = KeepTerminator.no, Range)(Range r)
+if ((hasSlicing!Range && hasLength!Range) ||
+    isSomeString!Range)
+{
+    import std.uni : lineSep, paraSep;
+    import std.conv : unsigned;
+
+    static struct Result
+    {
+    private:
+        Range _input;
+        alias IndexType = typeof(unsigned(_input.length));
+        enum IndexType _unComputed = IndexType.max;
+        IndexType iStart = _unComputed;
+        IndexType iEnd = 0;
+        IndexType iNext = 0;
+
+    public:
+        this(Range input)
+        {
+            _input = input;
+        }
+
+        static if (isInfinite!Range)
+        {
+            enum bool empty = false;
+        }
+        else
+        {
+            @property bool empty()
+            {
+                return iStart == _unComputed && iNext == _input.length;
+            }
+        }
+
+        @property Range front()
+        {
+            if (iStart == _unComputed)
```

Comment:
You won't be able to measure a difference for this one test in such a large function.

Functions consisting of essentially one `if` are the pits. The reader sees the beginning of a bulky `if` and goes, "boy that's some good long `if` statement, surely this function will do many wonderful interesting things long after this `if` is done". So the reader puts the `if` on the mental stack and goes on to read the large body of code inside the `if`, to just realize much later that all that was for naught - there's nothing after `if` so it needn't be remembered in the first place!

This is not good style, pure and simple. It's not reason enough to reject this code, but you'd do good to refactor it and anyone coming later would do good to refactor it.